src/utils/rpc_errors.ts

```diff
@@ -2,30 +2,50 @@
 import Mustache from 'mustache';
 import schema from '../generated/rpc_error_schema.json';
 import messages from '../res/error_messages.json';
-import * as CLASSMAP from '../generated/rpc_error_types';
-import { ServerError } from '../generated/rpc_error_types';
+import { TypedError } from './errors';
+import { ExecutionOutcomeWithIdView } from '../providers/provider';
 
-export * from '../generated/rpc_error_types';
+export class ServerError extends TypedError {
```

Comment:
I suggest merging `ServerError` and `ServerTransactionError` into `TransactionExecutionFailure` type with required `transaction_outcome` (for backward compatibility, if needed) and `context` fields.

Reply:
I also suggest renaming the file (`rpc_errors.ts`) or reorganizing it altogether (e.g. somehow merge it with the `transaction.ts`)

Reply:
Looks like a merge of these two classes is not possible with the current architecture, as well as part of the renaming that you have mentioned.
Basically, we are using `ServerError` for these type of errors:
`{"jsonrpc":"2.0","error":{"code":-32000,"message":"Server error","data":{"TxExecutionError":{"InvalidTxError":{"InvalidNonce":{"tx_nonce":2,"ak_nonce":28}}}}},"id":124}`

And `ServerTransactionError` for these type of errors:

`Response.result:
{
  status: { Failure: { ActionError: [Object] } },
  transaction: {},
  transaction_outcome: {},
  receipts_outcome: []
}`

Also, let's not rename `type` to `error_kind` in this PR because it would be a breaking change. Let's deliver this change in a separate PR after all changes in `nearcore`.